redis/pool.go

```diff
@@ -315,14 +312,10 @@ func (p *Pool) get(ctx context.Context) (*poolConn, error) {
 		if wait {
 			start = time.Now()
 		}
-		if ctx == nil {
-			<-p.ch
-		} else {
-			select {
-			case <-p.ch:
-			case <-ctx.Done():
-				return nil, ctx.Err()
-			}
+		select {
+		case <-p.ch:
+		case <-ctx.Done():
```

Comment:
This will cause a panic if ctx is nil e.g.
https://play.golang.org/p/F4NjaHeF4bi

Reply:
Thank you!
I reverted!